src/components/Header.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState, useRef, useEffect } from 'react';
+import { Button } from './Button';
+import { Todo } from '../types/Todo';
+import { USER_ID } from '../api/todos';
+import { addTodos } from '../api/todos';
+import { ErroMessage } from '../utils/errorMessages';
+
+type Props = {
+  todos: Todo[];
+  addTodo: (newTodo: Todo) => void;
+  handleErrorMessages: (newErrorMessage: ErroMessage) => void;
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+  setIsLoading: React.Dispatch<React.SetStateAction<number[]>>;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  completeAllTodos: () => void;
+  shouldFocusInput: boolean;
+  setShouldFocusInput: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  addTodo,
+  handleErrorMessages,
+  setTempTodo,
+  setIsLoading,
+  completeAllTodos,
+  shouldFocusInput,
+  setShouldFocusInput,
+}) => {
+  const [query, setQuery] = useState('');
+  const [disableInput, setDisableInput] = useState(false);
+  const inputElement = useRef<HTMLInputElement>(null);
+  const handleFocusInput = () => {
+    if (inputElement.current) {
+      inputElement.current.focus();
+    }
+  };
+
+  const handleSubmitForm = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (!query) {
+      handleErrorMessages(ErroMessage.EMPTY_TITLE);
+
+      return;
+    }
+
+    const newTodo = {
+      title: query.trim(),
+      completed: false,
+      userId: USER_ID,
+    };
+
+    try {
+      setDisableInput(true);
+
+      const tempTodo = {
+        title: query,
+        completed: false,
+        id: 0,
+        userId: USER_ID,
+      };
+
+      setTempTodo(tempTodo);
+      setIsLoading(prev => [...prev, tempTodo.id]);
+
+      const responseTodo = await addTodos(newTodo);
+
+      addTodo(responseTodo);
+
+      setQuery('');
+    } catch {
+      handleErrorMessages(ErroMessage.ADD);
+    } finally {
+      setDisableInput(false);
+      setTempTodo(null);
+      setShouldFocusInput(true);
+    }
+  };
+
+  useEffect(() => {
+    if (shouldFocusInput) {
+      handleFocusInput();
+    }
+  }, [handleFocusInput]);
```

Comment:
Please update dependencies array for the effect, or remove `shouldFocusInput` from the component at all, if it never changes